File: Programmers/[43162] 네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False for _ in range(n)]

    def dfs(computer):
        for i in range(n):
            if computers[computer][i]:
                if not visited[i]:
                    visited[i] = True
                    dfs(i)

    for i in range(n):
        if not visited[i]:
            visited[i] = True
            answer += 1
            dfs(i)

    return answer


if __name__ == "__main__":
    # n, computers = 3, [[1, 1, 0], [1, 1, 0], [0, 0, 1]]
    n, computers = 3, [[1, 1, 0], [1, 1, 1], [0, 1, 1]]
    # n, computers = 3, [[1, 0, 0], [0, 1, 0], [0, 0, 1]]
    print(solution(n, computers))

File: Programmers/[43162] 네트워크.java
import java.util.Queue;
import java.util.LinkedList;

class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] isVisited = new boolean[n];
        
        for (int i = 0; i < n; i++) {
            if (!isVisited[i]) {
                Queue<Integer> queue = new LinkedList<>();
                queue.offer(i);
                isVisited[i] = true;
                while (queue.size() > 0) {
                    int popped = queue.poll();
                    for (int j = 0; j < n; j++) {
                        if (computers[j][popped] == 1) {
                            if (!isVisited[j]) {
                                queue.offer(j);
                                isVisited[j] = true;
                            }
                        }
                    }
                }
                answer += 1;
            }
        }
        
        return answer;
    }
}